### PR TITLE
add test for PackageHelmChartTask

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: https://github.com/giantswarm/giantswarm/issues/5206

Add unit test for helm package task (`PackageHelmChartTask`).

1. create a chart
2. package the chart
3. check that the created helm chart archive contains what we expect

### CODE REVIEW

File: pack/pack_test.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+		expectedChartName      string
+		expectedChartVersion   string
+		expectedDeploymentName string
+		expectedDeploymentData string
+		expectedValues         string
+
+		setup func(chartDir string) error
+	}{
+		{
+			name:                   "case 0: test package chart",
```
Comment (MarcelMue) [inline pack/pack_test.go]: Would be nice to have a second test to check what happens if one of the inputs is missing. e.g. no `Chart.yaml`. Or no `values.yaml`.

**Comment by TheoBrigitte** [comment]
> LGTM. Does it already have multi-chart support?

This has no support for multi-chart, as multi-chart is built in the [`processHelmDir`](https://github.com/giantswarm/architect/blob/master/workflow/workflow.go#L315) function. I will link those together in a later PR.

File: pack/pack_test.go (hunk 1 above)
Comment (TheoBrigitte) [reply pack/pack_test.go]: added 2 more test:
* no chart, which yield `"chart metadata (Chart.yaml) missing"` error
* only Chart.yaml, which is the minimum valid chart.

File: pack/pack_test.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+				}
+
+				return createFiles(files)
+			},
+		},
+		{
+			name: "case 1: no chart",
+			errorMatcher: func(err error) bool {
+				expectedError := errors.New("chart metadata (Chart.yaml) missing")
+				return err.Error() == expectedError.Error()
```
Comment (TheoBrigitte) [inline pack/pack_test.go]: I'am not proud of this, but that's how the error is defined in helm: https://github.com/helm/helm/blob/master/pkg/chartutil/load.go#L200

**Pull request merged**